middleware/csrf.go

```diff
@@ -157,6 +165,9 @@ func CSRFWithConfig(config CSRFConfig) echo.MiddlewareFunc {
 			if config.CookieDomain != "" {
 				cookie.Domain = config.CookieDomain
 			}
+			if config.CookieSameSite != http.SameSiteDefaultMode {
```

Comment:
The article on the issue say this:
> Cookies with SameSite=None must also specify Secure, meaning they require a secure context.

So maybe it could be good to set cookie.Secure=true when SameSite=None

Reply:
Fixed

Reply:
@pafuent Maybe you can correct the testing strategy to take at least GO version 1.13 as a basis?
The http.SameSiteNoneMode constant appeared only in 1.13 version of GO(https://github.com/golang/go/issues/32546), which is why GitHub Actions throws an error when checking: https://github.com/labstack/echo/pull/1524/commits/dc147d9b974021cd79f98caf2f1159a66819533a

Reply:
This would mean this is a breaking change, as for vecho 4 we still consider Go v1.12 supported.
Do we rely on the actual implementation of SameSiteNoneMode in Golang, er just using the const for comparison?

Reply:
@lammel Are you suggesting to declare local constants that will be analogous in GoLang?

Reply:
Yes, that is what I meant. Go v1.12 will not have any changes in that regard anymore.
So in case we do not rely an additional functionality other than the constant I would prefer to do that.

Reply:
Unfortunately, there will be a conflict. 
The passed value in the config (`config.CookieSameSite`) is will be set as `cookie.SameSite`. Cookie, in turn, is a `http.Cookie` structure and an attempt to set a value from a local constant will result in an error.

Reply:
Maybe I'm missing something but maybe we can do something like this:

```go
package main

import (
	"fmt"
)

type SameSite int

const (
	SameSiteDefaultMode SameSite = iota + 1
	SameSiteLaxMode
	SameSiteStrictMode
)

const (
	SameSiteNoneMode SameSite = SameSiteStrictMode + 1
)

func main() {
	var ss SameSite
	ss = SameSiteLaxMode
	fmt.Println(ss)
	ss = SameSiteNoneMode
	fmt.Println(ss)
}
```

The tricky part is how to write it in order to select our version of the constant or the Go 1.13.
One option is to use build constraints. So only for Go 1.12 you define an Echo constant **SameSiteDefaultMode SameSite** with a value equal to **4**, for all the other versions, you define the same constant but equal to **http.SameSiteDefaultMode**

Something like this (please test it, I didn't had the time to do it)

samesite.go
```go
// +build !go1.12

package middleware

import (
	"net/http"
)

const (
	SameSiteNoneMode http.SameSite = http.SameSiteNoneMode
)
```

samesite_1_12.go
```go
// +build go1.12

package middleware

import (
	"net/http"
)

const (
	SameSiteNoneMode http.SameSite = 4
)
```